### ISSUE TITLE
different training loss between torchvision model loaded directly and downloaded model loaded

### ISSUE THREAD

**Issue by sanweiliti** [pytorch/vision#694]
Hi, I want to use pretrained resnet50 weights provided by torchvision (pytorch 0.4.1) for my model (called `fcrn`) which consists of a resnet50 (except the average pooling and fc layer) and an additional up-sampling decoder. I tried two different ways to load the pretrained resnet50 weights, and set the random seed to the same value, but got different training loss, anyone know why? (Since the random seed are the same, I think the training loss should be exactly the same under these two different loading methods). The weights are only loaded for the resnet50 part in my model, and the remaining part of my model are initialized with same methods in these two cases.

For the first method, I directly load weights from the torchvision model resnet50:
`model = fcrn(n_classes=n_classes).to(device)`
`resnet50 = torchvision.models.resnet50(pretrained=True)`
`initial_state_dict = resnet50.state_dict()`
`model.load_state_dict(initial_state_dict, strict=False)`

For the second method: I downloaded the pretrained resnet50 model from the link provided by this repo (https://download.pytorch.org/models/resnet50-19c8e357.pth):
`model = fcrn(n_classes=n_classes).to(device)`
`initial_state_dict = torch.load('resnet50-19c8e357.pth', map_location=lambda storage, loc: storage)`
`model.load_state_dict(initial_state_dict, strict=False)`

**Comment by fmassa** [comment]
The difference in training loss is expected.

Even if you set the same random seed everywhere (including in the dataloader workers), some of the algorithms that we use are non-deterministic.

You could try setting `torch.backends.cudnn.deterministic = True`, but it will make the code slower, and other parts of the codebase can also have non-deterministic implementations.
